src/imperative/cached_op.cc

```diff
@@ -705,8 +705,10 @@ void CachedOp::StaticRunOps(
           arg_shapes.emplace_back(ndinput->shape());
           arg_dtypes.emplace_back(ndinput->dtype());
         }
-        state.op_states[i] = createop[node.source->op()](
-            node.source->attrs, default_ctx, arg_shapes, arg_dtypes);
+        if (!config_.static_shape) {
```

Comment:
This is the only changed line to fix this PR. We should re-create op state if `static_shape` is false.